src/spartan_router.erl

```diff
@@ -13,30 +13,24 @@
 -spec(upstreams_from_questions(dns:questions()) -> ordsets:ordset(upstream())).
 upstreams_from_questions([#dns_query{name=Name}]) ->
     Labels = spartan_app:parse_upstream_name(Name),
-    lists:map(fun normalize_ip/1, find_upstream(Name, Labels));
```

Comment:
I won't categorize it as a flawed pattern. This function is to convert all different input types into something that erlang understands. I feel its a useful functionality and help us not to expose erlang types to users. You wouldn't expect/want an operator to know erlang internals.